### PR TITLE
[hotfix] Refactoring docker-entrypoint.sh to Use Arrays for Configuration Parameters Instead of Eval and String Concatenation

### PR DESCRIPTION
**Pull request by JunRuiLee**
Refactor the docker-entrypoint.sh script as per the suggestions in the comments of https://github.com/docker-library/official-images/pull/16430#issuecomment-2005337882

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -65,19 +65,19 @@ set_config_options() {
...
 
-    local config_params=""
+    local config_params=()
 
     while [ $# -gt 0 ]; do
         local key="$1"
         local value="$2"
-
-        config_params+=" -D${key}=${value}"
+
```
Comment (lincoln-lil) [inline docker-entrypoint.sh:73]: nit: remove spaces

**Review by lincoln-lil — APPROVED** [review]
+1 (refers to the discussion on https://github.com/docker-library/official-images/pull/16430)

**Comment by JunRuiLee** [comment]
Thanks @lincoln-lil for quick review, I've updated this pr accordingly, PTAL~

**Comment by lincoln-lil** [comment]
@JunRuiLee Thanks for the updates! LGTM, merging.

**Comment by lincoln-lil** [comment]
@JunRuiLee Could you also create a backport pr for dev-1.19?

**Pull request merged**